inference/core/workflows/core_steps/sinks/roboflow/model_monitoring_inference_aggregator/v1.py

```diff
@@ -0,0 +1,317 @@
+        "registration is needed, use `False` while debugging and if error handling is needed",
+        examples=[True],
+    )
+
+    @field_validator("frequency")
+    @classmethod
+    def ensure_frequency_is_correct(cls, value: Any) -> Any:
+        if isinstance(value, int) and value < 1:
+            raise ValueError("`frequency` cannot be lower than 1.")
+        return value
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="error_status", kind=[BOOLEAN_KIND]),
+            OutputDefinition(name="message", kind=[STRING_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.3.0,<2.0.0"
+
+
+@dataclass
+class Prediction(object):
+    class_name: str
+    confidence: float
+    inference_id: str
+    model_type: str
+
+
+class PredictionsAggregator(object):
+    _raw_predictions: List[Union[sv.Detections, dict]]
+
+    def __init__(self):
+        self._raw_predictions = []
+
+    def collect(self, value: Union[sv.Detections, dict]) -> None:
+        self._raw_predictions.append(value)
+
+    def _consolidate(self) -> List[Prediction]:
+        formatted_predictions = []
+        for detections in self._raw_predictions:
+            f = format_sv_detections_for_model_monitoring(detections)
+            formatted_predictions.extend(f)
+
+        class_groups = defaultdict(list)
+        for prediction in formatted_predictions:
+            class_name = prediction["class_name"]
+            class_groups[class_name].append(prediction)
+
+        representative_predictions = []
+        for class_name, predictions in class_groups.items():
+            predictions.sort(key=lambda x: x["confidence"], reverse=True)
+            representative_predictions.append(predictions[0])
+
+        return representative_predictions
+
+    def get_and_flush(self) -> List[Prediction]:
+        predictions = self._consolidate()
+        self._raw_predictions = []
+        return predictions
+
+
+class ModelMonitoringInferenceAggregatorBlockV1(WorkflowBlock):
+
+    def __init__(
+        self,
+        cache: BaseCache,
+        api_key: Optional[str],
+        background_tasks: Optional[BackgroundTasks],
+        thread_pool_executor: Optional[ThreadPoolExecutor],
+    ):
+        if api_key is None:
+            raise ValueError(
+                "ModelMonitoringInferenceAggregator block cannot run without Roboflow API key. "
+                "If you do not know how to get API key - visit "
+                "https://docs.roboflow.com/api-reference/authentication#retrieve-an-api-key to learn how to "
+                "retrieve one."
+            )
+        self._api_key = api_key
+        self._cache = cache
+        self._background_tasks = background_tasks
+        self._thread_pool_executor = thread_pool_executor
+        self._last_report_time_cache_key = "roboflow_model_monitoring_last_report_time"
```

Comment:
that would have a side effect for **all blocks of this type to have shared state of last report time** - is that desired end?

Reply:
* if the expectation is that it works at hosted, I would use redis with the following structure of keys `workflows:steps_cache:roboflow_core/model_monitoring_inference_aggregator@v1:{uuid4()}:last_report_time`